### ISSUE TITLE
incorrect tree reconstruction

### ISSUE THREAD

**Issue by vilterp** [vilterp/dist-tracing-view#10]
### Repro:

**Query:**
```sql
SELECT ol_number, count(*) FROM order_line GROUP BY ol_number
```

**Plan:**
[viewer](https://cockroachdb.github.io/distsqlplan/decode.html#eJykkkFLw0AQhe_-imVOLSxtNrZo9pTqqVgSqS0iEsKaHUMwyZadDVZK_rskOdQKSsXje7Nv-N4yB6iNxkhVSCCfQUDCYWdNhkTGdtbwYKn3ID0ORb1rXGcnHDJjEeQBXOFKBAkb9VLiGpVGO_WAg0anirJfa6xGm5ZFjenrW3hUwCFunGThDDhY807MotKS-UFwHXQ7yKmyZK6oUDLfn0_ms8uKIGk5mMYdUcipHEGKlp-Pu8hzi7lyxk7FKW3Y6UX0lEbxJo22q9UoFGPgcBtvo026jh8fRuNzeANv4omrioBDpfaswsrYD9YQasmEx-6Kmx-b-H9pskbamZrwtMX_v9NrEw6ocxzugExjM7y3JuuJBhn3ud7QSG6YikEs62HUdfkaFr-G_W_hpL34DAAA__8K6tgV)
![image](https://user-images.githubusercontent.com/7341/47666972-3b784c00-db7b-11e8-89cf-4f90fe88cbd6.png)

**Trace CSV:**
https://gist.github.com/vilterp/64c930c70896930c0564b77c86ea0251#file-tpcc-group-by-trace-csv

**Constructed tree:**
https://gist.github.com/vilterp/64c930c70896930c0564b77c86ea0251#file-tpcc-group-by-tree-json

**Screenshot:**
(the one above and to the right of the hash aggregator is its parent, a batch span, which starts after it)
![image](https://user-images.githubusercontent.com/7341/47666802-da507880-db7a-11e8-9f4f-d04fad172789.png)
I don't think this is actually its parent. Its actual parent should be the `Flow` span far above, the same parent as the `TableReader` span. I guess I'm still a bit skeptical that we can reconstruct the tree without having parent span IDs on each row.

**Comment by vilterp** [comment]
Talked with @BramGruneir; it seems we can't actually properly reconstruct the trace tree without having the the parent span id on each `start span` row. (Or returning the entire tree directly, as proposed in https://github.com/cockroachdb/cockroach/issues/31562)

**Comment by vilterp** [comment]
Here's an old commit (never merged) to add a `parent_span_idx` column to `crdb_internal.session_trace`: https://github.com/cockroachdb/cockroach/commit/f84bb6450df0c7a86d905b633e2045a69e8681e8

**Comment by BramGruneir** [comment]
Yep, I'm pretty sure I can still reconstruct the tree correctly based on the timestamps.  Will try to do just that.

But having the parent span would solve this at the source.

**Comment by vilterp** [comment]
Partially addressed by the above, but we need to output parent spans to be sure. I'll send a PR. (Tried to rebase my old one but stuff has moved around a lot)
